### DISCUSSION TITLE
How can I search message in channel by specific word or type?

### DISCUSSION THREAD

**Discussion by kkimwooo** [GetStream/stream-chat-flutter#768] [Q&A] [answered]
I want make emphasize effect to specific message in Channel Page.
But I don't know how to query message in channel by message type or specific word.

**Comment by imtoori** [comment]
hey @kkimwooo can you elaborate more? I'm not really getting what you want to do

**Comment by kkimwooo** [comment, marked as answer]
I wanted to find message in current channel.

I solved problem.
I used client.search() function to find specific message.

like this

`client
        .search(
          Filter.and(
            [
              Filter.equal('cid', streamChannel!.channel.cid.toString()),
            ],
          ),
          messageFilters: Filter.equal('customType', 'notice'),
        )
        .then((value) => value.results);
`
